app/assets/javascripts/components/upvote.es6.jsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * @prop discussion - discussion
+ * @prop response = response
+ * @prop user - current user
+ * @prop upvotes - total upvotes
+ */
+
+class Upvote extends React.Component {
+  constructor(props) {
+    super(props);
+    this._handleUpvote = this._handleUpvote.bind(this);
+    this._successfulSave = this._successfulSave.bind(this);
+    this.state = {
+      score: this.props.upvotes.length,
+      has_upvoted: false,
+      response: this.props.response
+    };
+  }
+
+  _handleUpvote() {
+    for (var i = 0; i < this.props.upvotes.length; i++) {
```

Comment:
I think we can move this for loop inside the if statement check "if(!this.state.has_upvoted)" - no need to loop through and set the state equal to true if it's already true